### ISSUE TITLE
Disk space low on jenkins controller's `jobs` directory

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#3285]
Disk was running at around 99% space used yesterday (It's a 2Tb file system) as alerted by Nagios

Top 20 disk hogs are as follows:
```
13337900	Test_openjdk17_hs_extended.openjdk_aarch64_mac_testList_1
13807988	build-scripts/jobs/release-openjdk21-pipeline/builds
17488432	build-scripts/jobs/openjdk22-pipeline/builds
19428212	build-scripts/jobs/openjdk20-pipeline/builds
20769276	GetNode
21553540	scott-fryer-adoptium-packages-linux-pipeline_new
26478320	build-scripts/jobs/release-openjdk20-pipeline/builds
29804056	build-scripts/jobs/openjdk18-pipeline/builds
32542916	build-scripts/jobs/openjdk19-pipeline/builds
33590700	build-scripts/jobs/release-openjdk8-pipeline/builds
36050596	build-scripts/jobs/openjdk17-pipeline/builds
36958568	openjdk_build_docker_multiarch
54007128	build-scripts/jobs/openjdk21-pipeline/builds
60749808	build-scripts/jobs/release/jobs
66742004	dependency_pipeline
70146480	build-scripts/jobs/release-openjdk17-pipeline/builds
74281632	build-scripts/jobs/release-openjdk11-pipeline/builds
133479468	build-scripts/jobs/openjdk8-pipeline/builds
211075888	build-scripts/jobs/openjdk11-pipeline/builds
263518372	build-scripts/jobs/jobs/jobs
```
I'm not sure what [GetNode](https://ci.adoptium.net/job/GetNode/) is for but it seems to have artifacts for jtreg etc. similar to [dependency_pipeline](https://ci.adoptium.net/view/Dependencies/job/dependency_pipeline/) It also hasn't run successfully for a while - last good run seems to have been [18 months ago](https://ci.adoptium.net/job/GetNode/428/). @sophia-guo Since GetNode seems to have been most recently edited by you do you know what's it's for, is it still required, and does it need to keep so many artifacts?

@steelhead31 Is your Linux packaging job still needed?

@andrew-m-leonard I think we need to consider what to do with all the release pipelines. At the moment we keep them all locked, but that's going to continually increase the disk space requirements here. We've now switched to the release-openjdkXX-pipelines jobs but some of the older releases from before the `release-` jobs came about are "locked" jobs in the `openjdkXX-pipeline` ones. We should probably also look at whether there are things unnecessarily locked in `build-scripts/jobs`

The openjdk_build_docker_multiarch is keeping the container images for old releases in the adoptopenjdk dockerhub account up to date, but it's keeping the logs indefinitely at about 40Mb/day and it's at close to 1000 runs hence the 36Gb of space in use. We should likely stop it retaining more than, say, a month's worth of logs to reclaim that space.

**Comment by smlambert** [comment]
GetNode seems to call the buildenv/jenkins/getDependency script (for which we have a separate job).  I have modified it to only keep 10 days of history, but would suspect it could be completely deleted, defer to Sophia's response.

**Comment by sxa** [comment]
Space on this file system expired during the January release cycle impacting the mac builds. Ref:
- https://ci.adoptium.net/job/build-scripts/job/jobs/job/release/job/jobs/job/jdk11u/job/jdk11u-release-mac-x64-temurin/16/console
- https://adoptium.slack.com/archives/C53GHCXL4/p1705669064638849

**Comment by smlambert** [comment]
re: https://github.com/adoptium/infrastructure/issues/3285#issuecomment-1900420126 - may also explain issues seen during release for TRSS being slow to get and display info from this Jenkins instance

**Comment by sxa** [comment]
There are 651 individual jobs which have `<keepLog>true</keepLog>`.

Will do a breakdown of how much space in in use by each of them so they can be reviewed appropriately. As an example the first one on the list was https://ci.adoptium.net/job/Test_openjdk11_j9_sanity.openjdk_aarch64_linux_xl/35/ from 2020 which I feel will be one that is no longer required (It's taking up 3Gb on its own)

**Comment by andrew-m-leonard** [comment]
https://ci.adoptium.net/job/dependency_pipeline/ does not have Discard old builds set, so all **1072 builds** is storing artifacts. @sophia-guo can we just keep the Last successful build ?

**Comment by andrew-m-leonard** [comment]
Id suggest we delete the out of support pipeline builds:
build-scripts/jobs/openjdk18-pipeline/builds
build-scripts/jobs/openjdk19-pipeline/builds
build-scripts/jobs/openjdk20-pipeline/builds

**Comment by andrew-m-leonard** [comment]
i’d also suggest we could delete the old build jobs from here: https://ci.adoptium.net/job/build-scripts/job/jobs/
for: jdk20u, jdk21





[11:25](https://adoptium.slack.com/archives/DLMGK43JR/p1706786717880509)
ditto for evaluation: https://ci.adoptium.net/job/build-scripts/job/jobs/job/evaluation/job/jobs/

**Comment by sxa** [comment]
Removing the following which appear to be hangovers from 2021 and appear to be "workspace" style directories of transient information which doesn't need to be there:
```
jenkins-hetzner-ubuntu2004:/home/jenkins/.jenkins/jobs/build-scripts$ du -sk *openjdk* | grep -v tmp
12866244        openjdk11-pipeline
11480852        openjdk11-pipeline@2
3538016 openjdk11-pipeline@3
275900  openjdk11-pipeline@4
7368    openjdk11-pipeline@5
3156    openjdk11-pipeline@libs
7412    openjdk11-pipeline@script
8418440 openjdk16-pipeline
7125692 openjdk16-pipeline@2
3095692 openjdk16-pipeline@3
561028  openjdk16-pipeline@4
1688    openjdk16-pipeline@libs
7512    openjdk16-pipeline@script
8375924 openjdk17-pipeline
5656448 openjdk17-pipeline@2
1592404 openjdk17-pipeline@3
1632    openjdk17-pipeline@libs
7512    openjdk17-pipeline@script
7063876 openjdk8-pipeline
7076332 openjdk8-pipeline@2
2593056 openjdk8-pipeline@3
278172  openjdk8-pipeline@4
2704    openjdk8-pipeline@libs
7256    openjdk8-pipeline@script
6560    weekly-openjdk11-pipeline@script
8616    weekly-openjdk15-pipeline@script
6560    weekly-openjdk16-pipeline@script
6560    weekly-openjdk17-pipeline@script
6560    weekly-openjdk8-pipeline@script
```
I'm creating a backup to keep for a week before deleting them, but this is about 80Gb in total.

**Comment by steelhead31** [comment]
Running this script shows disk space usage increased by c.100Gb in a single day, I suspect this is a good example of our burn rate

```
#!/bin/bash

# Specify the target directory
target_directory="/home/jenkins/.jenkins/jobs"

# Specify the date for which you want to calculate disk space usage
target_date="2024-02-08"

# Convert the target date to timestamp
target_timestamp=$(date -d "$target_date" +%s)

# Calculate disk space usage for the specified date
disk_space_used=$(find "$target_directory" -type f -newermt "$target_date" -exec du -c {} + | grep "total" | awk '{sum += $1} END {print sum}')

# Print the result
echo "Disk space used on $target_date: $disk_space_used KB"
``

**Comment by sxa** [comment]
> Running this script shows disk space usage increased by c.100Gb in a single day,

To be clear, the disk usage didn't really increase by 100Gb/day - that's the size of new things that have been generated. The jenkins "discard old builds" option means that older ones will be getting deleted alongside these new ones.

**Comment by sxa** [comment]
> Id suggest we delete the out of support pipeline builds: build-scripts/jobs/openjdk18-pipeline/builds build-scripts/jobs/openjdk19-pipeline/builds build-scripts/jobs/openjdk20-pipeline/builds

👍🏻 on this - We've got an issue with the eclipse-mirror job at the moment and once that is resolved and I've mirrored the latest 18 and 19 releases I'll look at clearing those out from jenkins.

**Comment by sxa** [comment]
I've pushed out the jdk-18.0.2.1+1 and jdk-19.02+7 builds to the Eclipse mirror, so we have those retained properly.

Deep dive into openjdk18-pipeline which using 29Gb in total:
The following jobs are taking more the 1Gb so have some artifacts retained (based on `du -sk build-scripts/jobs/openjdk18-pipeline/builds/* | awk '$1>1000000'`
<details>
<summary>openjdk18-pipeline (29Gb)</summary>

Job | Size | Notes
---|---|---
#151 | 1.8Gb | jdk-18+36
#153 | 2.8Gb | jdk-18+36 (subset of platforms - presumably failed from 151)
#170 | 4.8Gb | jdk-18.0.1+10 (#171 = respin mac/x64 only)
#229 | 5.9Gb | jdk-18.0.2+9
#250 | 5.9Gb | jdk-18.0.2.1+1
#272 | 1.0Gb | xLinux and a64Linux only
#273 | 6.3Gb | Latest build (although not same SHA as 18.0.2.1+1)
</details>

<details>
<summary>openjdk19-pipeline (32Gb)</summary>

Job | Size | Notes
---|---|---
#90 | 4.7Gb | Reproducible test
#216 | 1.7Gb | jdk-19+36 (x64 primaries)
#217 | 4.5Gb | jdk-19+36
#237 | 5.8Gb | jdk-19.0.1+10
#311 | 4.1Gb | Seems to be retaining 3 ...
#312  | 3.3Gb | Second last (retain )
#313 | 5.7Gb | Latest
</details>

<details>
<summary>openjdk20-pipeline (19Gb)</summary>

Job | Size | Notes
---|---|---
#246 | 4.7Gb | Last successful build
#249 | 1.7Gb | jdk-19+36 (x64 primaries)
#250 | 4.5Gb | jdk-19+36
</details>

<details>
<summary>openjdk17-pipeline (42Gb)</summary>

Job | Size | Notes
---|---|---
#348 | 4.2Gb | jdk-17.0.3+7
#462 | 5.3Gb | jdk-17.0.4+7
#463 | 5.4Gb | jdk-17.0.4+8 (+win32 in #465)
#480 | 5.8Gb | jdk-17.0..1+1
#539 | 5.9Gb | jdk-17.0.5+8 (549 was Alpine/aarch64)
#873 | 3.8Gb | Last green (config=2 retained)
#895 | 6.5Gb | Second last (Red status)
#896 | 4.9Gb | Last (Black status)

Earlier pipelines exist, but not with artifacts
</details>

<details>
<summary>openjdk11-pipeline (206Gb)</summary>
Generated from: du -sk build-scripts/jobs/openjdk11-pipeline/builds/* | awk '$1>1000000' | while read A B; do NUM=`basename $B`; echo $NUM \| $A \| `grep "<displayName>" build-scripts/jobs/openjdk11-pipeline/builds/$NUM/build.xml | sed -e "s/^[^>]*>//" -e "s/<.*$//"`; done | sort -n

Job | Size (bytes) | Notes
---|---|---
414 | 2291736 | jdk-11.0.3+7_openj9-0.14.0
419 | 1215672 | jdk-11.0.3+7
453 | 2551692 | jdk-11.0.3+7_openj9-0.14.2
457 | 2551752 | jdk-11.0.3+7_openj9-0.14.3
508 | 1120232 | jdk-11.0.4+11_adopt
509 | 1270648 | jdk-11.0.4+11_adopt
511 | 3009224 | jdk-11.0.4+11_openj9-0.15.1
513 | 1120252 | jdk-11.0.4+11
514 | 1046092 | jdk-11.0.4+11
559 | 1153680 | jdk-11.0.4+11_openj9-0.15.1
564 | 1382980 | jdk-11.0.4+11_openj9-0.15.1
613 | 2901304 | jdk-11.0.5+10_openj9-0.17.0
619 | 1212616 | jdk-11.0.5+10
620 | 1414952 | jdk-11.0.5+10
628 | 1852104 | jdk-11.0.5+10_openj9-0.17.0
704 | 1628240 | jdk-11.0.6+10
706 | 1257828 | jdk-11.0.6+10
709 | 2900008 | jdk-11.0.6+10_openj9-0.18.0
720 | 3167056 | jdk-11.0.6+10_openj9-0.18.1
856 | 3085644 | jdk-11.0.7+10
857 | 4118888 | jdk-11.0.7+10_openj9-0.20.0
865 | 3301324 | jdk-11.0.7+10_openj9-0.20.0
877 | 2447092 | jdk-11.0.7+10_openj9-0.20.0
889 | 1042764 | jdk-11.0.7+10_openj9-0.20.0
1083 | 2436716 | jdk-11.0.8+10
1085 | 4333592 | jdk-11.0.8+10_openj9-0.21.0
1221 | 4998900 | jdk-11.0.9+11_openj9-0.23.0
1223 | 2494756 | jdk-11.0.9+11
1225 | 2227884 | jdk-11.0.9+11
1227 | 1035056 | jdk-11.0.9+11_openj9-0.23.0
1231 | 1157228 | jdk-11.0.9+11_openj9-0.23.0
1233 | 1884080 | jdk-11.0.9+11
1254 | 1773976 | jdk-11.0.9.1+1
1261 | 2907508 | jdk-11.0.9.1+1
1283 | 1157568 | jdk-11.0.9+11_openj9-0.23.0
1355 | 5557900 | jdk-11.0.10+9_openj9-0.24.0
1357 | 2500628 | jdk-11.0.10+9
1493 | 5518372 | jdk-11.0.11+8
1495 | 5518716 | jdk-11.0.11+9
1496 | 3220672 | jdk-11.0.11+9_openj9-0.26.0
1597 | 5285452 | jdk-11.0.12+7
1690 | 4297860 | jdk-11.0.13+7
1698 | 6255956 | jdk-11.0.13+8
1795 | 6764300 | jdk-11.0.14+9
1817 | 6764384 | jdk-11.0.14.1+1
1900 | 7251324 | jdk-11.0.15+8
1910 | 7252296 | jdk-11.0.15+10
2026 | 7136784 | jdk-11.0.16+7
2028 | 7164468 | jdk-11.0.16+8
2043 | 6121708 | jdk-11.0.16.1+1
2121 | 6667208 | jdk-11.0.17+8
2564 | 7279856 |
2669 | 10796296 |
2670 | 3366408 |

Earlier pipelines exist, but not with artifacts
</details>

<details>
<summary>openjdk8-pipeline (122Gb)</summary>

Job | size (bytes) | 
---|---|---
539 | 2004044 | jdk8u212-b03_openj9-0.14.0
643 | 2324344 | jdk8u222-b10_openj9-0.15.1
728 | 2172108 | jdk8u232-b09_openj9-0.17.0
750 | 1237172 | jdk8u232-b09_openj9-0.17.0
751 | 1381156 | jdk8u232-b09_openj9-0.17.0
830 | 1655296 | jdk8u242-b08
836 | 2663076 | jdk8u242-b08_openj9-0.18.1
990 | 3545120 | jdk8u252-b09_openj9-0.20.0
1003 | 1976000 | jdk8u252-b09_openj9-0.20.0
1170 | 1503088 | jdk8u262-b10
1172 | 3315252 | jdk8u262-b10_openj9-0.21.0
1187 | 1655092 | jdk8u265-b01
1188 | 2752920 | jdk8u265-b01_openj9-0.21.0
1309 | 1230560 | jdk8u272-b10
1311 | 3370412 | jdk8u272-b10_openj9-0.23.0
1315 | 1365480 | jdk8u272-b10
1356 | 1630644 | jdk8u275-b01_openj9-0.23.0
1460 | 3414276 | jdk8u282-b08_openj9-0.24.0
1462 | 1657468 | jdk8u282-b08
1601 | 1769932 | jdk8u292-b09
1603 | 1770064 | jdk8u292-b10
1604 | 2828460 | jdk8u292-b10_openj9-0.26.0
1672 | 2408036 | jdk8u292-b10
1715 | 2329912 | jdk8u302-b08
1827 | 2776224 | jdk8u312-b07
1932 | 2336224 | jdk8u322-b05-SXAPRE
1939 | 2238680 | jdk8u322-b05-SXAAQA
1941 | 2987104 | jdk8u322-b06
2087 | 3006336 | jdk8u332-b09
2202 | 2691196 | jdk8u342-b07
2212 | 3020520 | jdk8u345-b01
2249 | 3094472 | jdk8u352-b05-ea
2252 | 2728916 | jdk8u352-b05_ea
2254 | 2358324 |
2258 | 2143516 | jdk8u352-b05_ea (multi)
2322 | 2973544 | jdk8u352-b08
2476 | 4485940 |
2767 | 3329488 |
2859 | 2803624 |
</details>

<details>
<summary>release-openjdk8-pipeline (36Gb)</summary>

Job | Size | Notes
---|---|---
3 | 2531796 | jdk8u362-b06
5 | 2765316 | jdk8u362-b06
6 | 3093056 | jdk8u362-b06
7 | 2802032 | jdk8u362-b09
14 | 1767692 | jdk8u372-b06
15 | 3094524 | jdk8u372-b07
19 | 3095480 | jdk8u382-b03
20 | 3097024 | jdk8u382-b03
22 | 3100668 | jdk8u382-b05
27 | 2404220 | jdk8u392-b07
28 | 3108768 | jdk8u392-b08
37 | 2438792 | jdk8u402-b06

</details>

<details>
<summary>release-openjdk11-pipeline (80Gb)</summary>

Job | Size | Notes
---|---|---
1 | 7231564 | jdk-11.0.18+10
8 | 7230652 | jdk-11.0.19+6
9 | 6898168 | jdk-11.0.19+6
10 | 6897356 | jdk-11.0.19+7
12 | 6054832 | jdk-11.0.20+6
13 | 6920588 | jdk-11.0.20+8
15 | 7258412 | jdk-11.0.20.1+1
16 | 6567988 | jdk-11.0.20.1+1
17 | 4446640 | jdk-11.0.21+8
18 | 5879440 | jdk-11.0.21+8
19 | 7282912 | jdk-11.0.21+9
36 | 7261420 | jdk-11.0.22+7
39 | 1025316 | jdk-11.0.22+7_Mac_aarch64_ld_classic_linker

</details>


<details>
<summary>release-openjdk17-pipeline (77Gb)</summary>

Job | Size | Notes
---|---|---
8 | 4057804 | jdk-17.0.6+8
10 | 3887640 | jdk-17.0.6+8
12 | 3845936 | jdk-17.0.6+8
16 | 5417944 | jdk-17.0.6+8
17 | 4391964 | jdk-17.0.6+10
23 | 4840488 | jdk-17.0.7+6
24 | 5419768 | jdk-17.0.7+6
25 | 5420132 | jdk-17.0.7+7
31 | 5455628 | jdk-17.0.8+5
32 | 5456292 | jdk-17.0.8+7
33 | 5456376 | jdk-17.0.8.1+1
34 | 5457700 | jdk-17.0.8.1+1
35 | 3571952 | jdk-17.0.9+7
37 | 5439692 | jdk-17.0.9+9
51 | 5425028 | jdk-17.0.10+7
52 | 4933724 | jdk-17.0.10+7

</details>


<details>
<summary>release-openjdk19-pipeline (5.6Gb)</summary>

Job | Size | Notes
---|---|---
9 | 5790272 | jdk-19.0.2+7

</details>


<details>
<summary>release-openjdk20-pipeline (26Gb)</summary>

Job | Size | Notes
---|---|---
1 | 5286300 | jdk-20+35
3 | 5286792 | jdk-20+35
5 | 5289180 | jdk-20+36
9 | 5289768 | jdk-20.0.1+9
10 | 5323856 | jdk-20.0.2+9

</details>

<details>
<summary>release-openjdk21-pipeline (19Gb)</summary>

Job | Size | Notes
---|---|---
1 | 5286300 | jdk-20+35
3 | 5286792 | jdk-20+35
5 | 5289180 | jdk-20+36
9 | 5289768 | jdk-20.0.1+9
10 | 5323856 | jdk-20.0.2+9

</details>

**Comment by sxa** [comment]
openjdk8-pipeline cleared. I have only removed the binary artifacts (msi, zip, tar.gz) with the following command: `find build-scripts/jobs/openjdk8-pipeline -name OpenJDK8U\* -print | grep -E 'msi$|zip$|tar.gz$' | grep -v x64_linux | grep -v ` | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk8-pipeline.txt`

Notes:

- I'm keeping the metadata around, hence the file type filter - sbom, gpg signatures and SHA sums are retained.
- The command above should have had `grep -E 'msi$|zip$|pkg$|tar.gz$'` so I subsequently ran it on `.pkg` files only
- I felt more comfortable starting with one of the non-current pipelines instead of the `release-` prefixed ones
- The `grep -v 2859` was because that is the number of the last successful build of the pipeline and I wanted to leave that in place. Not a major problem since they will get replaced soon but I didn't want to break anyone's workflows that might be pulling from it.
- I've retained x64_linux for no rational reason other than I feel more comfortable leaving one in place for now (currently all of those including debug/testimages are retained)2859

The amount of space used for `openjdk8-pipeline` was 121Gb. It is now 16Gb. Each job is taking about 300Mb of which all but about 3Kb are the x64_linux files.

This is the log of deleted artifacts (includes the original and `.pkg` run concatenated): 
[deleted-artifacts-openjdk8-pipeline.txt.gz](https://github.com/adoptium/infrastructure/files/14224782/deleted-artifacts-openjdk8-pipeline.txt.gz)

For subsequent runs, this would be the correct command to use, adjusted to replace `openjdk8-pipeline` and `2859` as appropriate: `find build-scripts/jobs/openjdk8-pipeline -name OpenJDK8U\* -print | grep -v 2859 | grep -E 'msi$|zip$|pkg$|tar.gz$' | grep -v x64_linux | grep -v sources | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk8-pipeline.txt` (Note - this has been adjusted to retain the sources archive too)

**Comment by sxa** [comment]
Likely as a result of the problems with low disk space on this file system recently some files are truncated e.g. `Test_openjdk21_hs_sanity.openjdk_ppc64_aix_rerun/builds/2/build.xml`. This was found during https://github.com/adoptium/infrastructure/issues/3376 - we should check for occurrences of similar references in the jenkins jobs and mediate them (probably by deleting the jobs). Noting that the referenced file is exactly 32768 bytes which is consistent with it hitting a disk full message and being unable to allocate another block.

**Comment by sxa** [comment]
Pruned the following:
- openjdk18-pipeline: `find build-scripts/jobs/openjdk18-pipeline -name OpenJDK18U\* -print | grep -vw 250 | grep -vw 273 | grep -E 'msi$|zip$|pkg$|tar.gz$' | grep -v x64_linux | grep -v sources | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk18-pipeline.txt`
- openjdk19-pipeline: `find build-scripts/jobs/openjdk19-pipeline -name OpenJDK19U\* -print | grep -vw 313 | grep -E 'msi$|zip$|pkg$|tar.gz$' | grep -v x64_linux | grep -v sources | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk19-pipeline.txt`
- openjdk20-pipeline: `find build-scripts/jobs/openjdk20-pipeline -name OpenJDK20U\* -print | grep -vw 246 | grep -E 'msi$|zip$|pkg$|tar.gz$' | grep -v x64_linux | grep -v sources | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk20-pipeline.txt`

Those have given us 396Gb on the file system.

- openjdk11-pipeline: `find build-scripts/jobs/openjdk11-pipeline -name OpenJDK11U\* -print | grep -vw 1496 | grep -vw 1495 | grep -vw 2121 | grep -vw 2687 | grep -E 'msi$|zip$|pkg$|tar.gz$' | grep -v x64_linux | grep -v sources | xargs rm -v 2>&1 | tee $HOME/deleted-artifacts-openjdk11-pipeline.txt`
  - I've retained 1495, 1496 (Last openj9 and the equivalent adopt 11.0.11+9, 2121 (latest released on this pipeline 11.0.17+8 and 2687 (latest pipeline at the time of writing)

That means we're now up to 550Gb. I'm going to stop now although none of the release-openjdkXX-pipeline ones have been cleared.

Space in use by each pipeline is now:
```
jenkins@jenkins-hetzner-ubuntu2004:~/jobs$ du -sh build-scripts/jobs/openjdk*pipeline
39G	build-scripts/jobs/openjdk11-pipeline
8.6G	build-scripts/jobs/openjdk17-pipeline
15G	build-scripts/jobs/openjdk18-pipeline
14G	build-scripts/jobs/openjdk19-pipeline
8.8G	build-scripts/jobs/openjdk20-pipeline
56G	build-scripts/jobs/openjdk21-pipeline
14G	build-scripts/jobs/openjdk22-pipeline
13G	build-scripts/jobs/openjdk23-pipeline
17G	build-scripts/jobs/openjdk8-pipeline
```
I've got a little bit of checking to do on some of my locked 21 pipelines before clearing that one out.

**Comment by sxa** [comment]
Here are the current biggest space hog locations after performing the above cleanup:
```
12936736	build-scripts/jobs/openjdk23-pipeline/builds
14065000	build-scripts/jobs/openjdk22-pipeline/builds
14521308	build-scripts/jobs/openjdk19-pipeline/builds
14937348	build-scripts/jobs/openjdk18-pipeline/builds
15460760	adoptium-packages-linux-pipeline_new
17387720	build-scripts/jobs/openjdk8-pipeline/builds
17405984	build-scripts/jobs/openjdk21-pipeline/builds
19310112	build-scripts/jobs/release-openjdk21-pipeline/builds
26478320	build-scripts/jobs/release-openjdk20-pipeline/builds
37150212	build-scripts/jobs/release-openjdk8-pipeline/builds
40050756	build-scripts/jobs/openjdk11-pipeline/builds
49409832	build-scripts/jobs/release/jobs
80506272	build-scripts/jobs/release-openjdk17-pipeline/builds
83512772	build-scripts/jobs/release-openjdk11-pipeline/builds
94298548	dependency_pipeline
282769152	build-scripts/jobs/jobs/jobs
```
While there could be more cleanup work here (maybe some of the individual platform builds have things retained unnecessarily) I'll consider this complete for now. I'll raise a separate issue for [dependency_pipeline](https://ci.adoptium.net/job/dependency_pipeline/) (Noting that we have an issue for sorting out aspects of [build.getDependency](https://ci.adoptium.net/view/all/job/build.getDependency) although the space usage of that seems to have been mitigated)
